test/e2e/case38_install_operator_test.go

```diff
@@ -1,126 +1,518 @@
+		polName string,
+		wantNonCompliant bool,
+		expectedRelatedObjs []policyv1.RelatedObject,
+		expectedCondition metav1.Condition,
+		expectedEventMsgSnippet string,
+	) {
+		checkFunc := func(g Gomega) {
+			GinkgoHelper()
 
-			It("Should create the OperatorGroup", func() {
-				og := utils.GetWithTimeout(clientManagedDynamic, gvrOperatorGroup,
-					case38OgName, case38OpPolicyNS, true, defaultTimeoutSeconds)
+			unstructPolicy := utils.GetWithTimeout(clientManagedDynamic, gvrOperatorPolicy, polName,
+				opPolTestNS, true, opPolTimeout)
 
-				Expect(og).NotTo(BeNil())
-			})
+			policyJSON, err := json.MarshalIndent(unstructPolicy.Object, "", "  ")
+			g.Expect(err).NotTo(HaveOccurred())
 
-			It("Should create the Subscription from the spec", func() {
-				sub := utils.GetWithTimeout(clientManagedDynamic, gvrSubscription,
-					case38SubscriptionName, case38OpPolicyNS, true, defaultTimeoutSeconds)
+			GinkgoWriter.Printf("Debug info for failure.\npolicy JSON: %s\nwanted related objects: %+v\n"+
+				"wanted condition: %+v\n", string(policyJSON), expectedRelatedObjs, expectedCondition)
 
-				Expect(sub).NotTo(BeNil())
-			})
+			policy := policyv1beta1.OperatorPolicy{}
+			err = json.Unmarshal(policyJSON, &policy)
+			g.Expect(err).NotTo(HaveOccurred())
 
-			It("Should become Compliant", func() {
-				OpPlc := utils.GetWithTimeout(clientManagedDynamic, gvrOperatorPolicy,
-					case38OpPolicyName, case38OpPolicyNS, true, defaultTimeoutSeconds)
+			if wantNonCompliant {
+				g.Expect(policy.Status.ComplianceState).To(Equal(policyv1.NonCompliant))
+			}
 
-				Expect(utils.GetComplianceState(OpPlc)).To(Equal("Compliant"))
-			})
+			matchingRelated := policy.Status.RelatedObjsOfKind(expectedRelatedObjs[0].Object.Kind)
+			g.Expect(matchingRelated).To(HaveLen(len(expectedRelatedObjs)))
 
-			It("Should have installed the operator", func() {
-				deployment := utils.GetWithTimeout(clientManagedDynamic, gvrDeployment,
-					case38DeploymentName, case38OpPolicyNS, true, defaultTimeoutSeconds*2)
+			for _, expectedRelObj := range expectedRelatedObjs {
+				foundMatchingName := false
+				unnamed := expectedRelObj.Object.Metadata.Name == ""
 
-				Expect(deployment).NotTo(BeNil())
-			})
+				for _, actualRelObj := range matchingRelated {
+					if unnamed || actualRelObj.Object.Metadata.Name == expectedRelObj.Object.Metadata.Name {
+						foundMatchingName = true
+						g.Expect(actualRelObj.Compliant).To(Equal(expectedRelObj.Compliant))
+						g.Expect(actualRelObj.Reason).To(Equal(expectedRelObj.Reason))
+					}
+				}
+
+				g.Expect(foundMatchingName).To(BeTrue())
+			}
+
+			idx, actualCondition := policy.Status.GetCondition(expectedCondition.Type)
+			g.Expect(idx).NotTo(Equal(-1))
+			g.Expect(actualCondition.Status).To(Equal(expectedCondition.Status))
+			g.Expect(actualCondition.Reason).To(Equal(expectedCondition.Reason))
+			g.Expect(actualCondition.Message).To(Equal(expectedCondition.Message))
+
+			g.Expect(utils.GetMatchingEvents(
```

Comment:
Isn't it better to wrap it with Eventually? it may cause flaky issue

Reply:
Yes - this function is now called in an `Eventually`, and then also a `Consistently`; the idea is to check that it occurs at some point, and then when it happens, it is the "final" state of the resource. Without the `Consistently`, a bug I had in the code was causing it to pass the tests incorrectly sometimes. With the `Consistently`, the bug was always causing a failure (until I fixed it)